### PR TITLE
feat(dashboard): chat step editor and preview

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

Chat step editor and preview.

### Screenshots

https://github.com/user-attachments/assets/37adcac3-e1cc-46e3-860b-6c9a2e486e85

### CODE REVIEW

File: apps/dashboard/src/components/workflow-editor/steps/chat/chat-editor-preview.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+type ChatEditorPreviewProps = {
+  workflow: WorkflowResponseDto;
+  step: StepDataDto;
+  formValues: Record<string, unknown>;
+};
+
+const extensions = [loadLanguage('json')?.extension ?? []];
+
+export const ChatEditorPreview = ({ workflow, step, formValues }: ChatEditorPreviewProps) => {
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/steps/chat/chat-editor-preview.tsx:31]: It would be nice if we can refactor some of the previews and commonize it within one component. I tried that but it seems like more work and adjustments are needed for the in-app.

File: apps/dashboard/src/components/workflow-editor/steps/chat/chat-preview.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { RiSendPlane2Fill } from 'react-icons/ri';
+import { ChannelTypeEnum, ChatRenderOutput, GeneratePreviewResponseDto } from '@novu/shared';
+
+import { LogoCircle } from '@/components/icons';
+import { Skeleton } from '@/components/primitives/skeleton';
+import { cn } from '@/utils/ui';
+
+export const ChatPreview = ({
```
Comment (LetItRock) [inline apps/dashboard/src/components/workflow-editor/steps/chat/chat-preview.tsx:8]: the preview used in the editor and as a mini preview

File: apps/dashboard/src/components/workflow-editor/steps/chat/chat-editor.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { getComponentByType } from '@/components/workflow-editor/steps/component-utils';
+import { TabsSection } from '@/components/workflow-editor/steps/tabs-section';
+
+const bodyKey = 'body';
+
+type ChatEditorProps = { uiSchema: UiSchema };
+
+export const ChatEditor = (props: ChatEditorProps) => {
+  const { uiSchema } = props;
+  const { [bodyKey]: body } = uiSchema?.properties ?? {};
```
Comment (ChmaraX) [inline apps/dashboard/src/components/workflow-editor/steps/chat/chat-editor.tsx]:
```suggestion
  const { body } = uiSchema.properties ?? {};
```

File: apps/dashboard/src/components/workflow-editor/steps/push/push-editor.tsx [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
+import { TabsSection } from '@/components/workflow-editor/steps/tabs-section';
 
 const subjectKey = 'subject';
 const bodyKey = 'body';
 
 type PushEditorProps = { uiSchema: UiSchema };
+
 export const PushEditor = (props: PushEditorProps) => {
   const { uiSchema } = props;
   const { [bodyKey]: body, [subjectKey]: subject } = uiSchema?.properties ?? {};
```
Comment (ChmaraX) [inline apps/dashboard/src/components/workflow-editor/steps/push/push-editor.tsx]: I know its not part of this PR just noticed here. 
```suggestion
  const { body, subject } = uiSchema.properties ?? {};
```

**Review by ChmaraX — APPROVED** [review]

**Review by BiswaViraj — APPROVED** [review]
